src/WebJobs.Script/Extensions/ScriptLoggingBuilderExtensions.cs

```diff
@@ -26,9 +29,14 @@ public static ILoggingBuilder AddDefaultWebJobsFilters(this ILoggingBuilder buil
             return builder;
         }
 
-        private static bool Filter(string category, LogLevel actualLevel, LogLevel minLevel)
+        internal static bool Filter(string category, LogLevel actualLevel, LogLevel minLevel)
+        {
+            return actualLevel >= minLevel && IsFiltered(category);
+        }
+
+        private static bool IsFiltered(string category)
         {
-            return actualLevel >= minLevel && ScriptConstants.SystemLogCategoryPrefixes.Where(p => category.StartsWith(p)).Any();
+            return _filteredCategoryCache.GetOrAdd(category, c => ScriptConstants.SystemLogCategoryPrefixes.Where(p => category.StartsWith(p)).Any());
```

Comment:
Cache, to avoid doing a startswith string comparison for each category (currently there are 4) on every log statement.